### PR TITLE
Implement options for ZRANGE, ZRANGEBYSCORE, ZREVRANGE, ZREVRANGEBYSCORE

### PR DESCRIPTION
**Pull request by brstgt**

### CODE REVIEW

File: src/RedisSortedSet.php [hunk 1]
```diff
@@ -42,9 +42,14 @@ public function add(array $data): Promise
      *
      * @return Promise
      */
-    public function getRange(int $start, int $end): Promise
+    public function getRange(int $start, int $end, RangeOptions $options = null): Promise
```
Comment (kelunik) [inline src/RedisSortedSet.php:45]:
```suggestion
    public function getRange(int $start, int $end, ?RangeOptions $options = null): Promise
```

Comment (kelunik) [inline src/RedisSortedSet.php:45]: Why is it `RangeOptions` instead of `RangeByScoreOptions` here?

Comment (brstgt) [reply src/RedisSortedSet.php:45]: Actually a valid questions. I guess i was mislead by the interface of the redis extension by which I modeled those options while the latest redis docs show a different definition where ZRANGE has even more options:
https://redis.io/commands/zrange/ vs https://redis.io/commands/zrangebyscore/

> Starting with Redis 6.2.0, this command can replace the following commands: [ZREVRANGE](https://redis.io/commands/zrevrange), [ZRANGEBYSCORE](https://redis.io/commands/zrangebyscore), [ZREVRANGEBYSCORE](https://redis.io/commands/zrevrangebyscore), [ZRANGEBYLEX](https://redis.io/commands/zrangebylex) and [ZREVRANGEBYLEX](https://redis.io/commands/zrevrangebylex).

Do you have any opinion on how to deal with this protocol evolution? I can adjust the MR then.

Comment (kelunik) [reply src/RedisSortedSet.php:45]: I'd probably go with RangeByScoreOptions here for now, as the other options can be used with the specialized commands?

Comment (brstgt) [reply src/RedisSortedSet.php]: I pushed another commit. WDYT?

File: src/RangeByScoreOptions.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+        return $clone;
+    }
+
+    public function isWithScores(): bool
+    {
+        return $this->withScores;
+    }
+
+    public function withOffset(int $offset, int $count): self
```
Comment (kelunik) [inline src/RangeByScoreOptions.php:27]: Shouldn't this be called `withLimit` instead?

File: src/RedisSortedSet.php [hunk 3]
```diff
@@ -64,9 +74,14 @@ public function getReverseRange(int $start, int $end): Promise
      *
      * @return Promise
      */
-    public function getRangeByScore(float $min, float $max): Promise
+    public function getRangeByScore(float $min, float $max, RangeByScoreOptions $options = null): Promise
```
Comment (kelunik) [inline src/RedisSortedSet.php:77]:
```suggestion
    public function getRangeByScore(float $min, float $max, ?RangeByScoreOptions $options = null): Promise
```

File: src/RedisSortedSet.php [hunk 4]
```diff
@@ -75,9 +90,14 @@ public function getRangeByScore(float $min, float $max): Promise
      *
      * @return Promise
      */
-    public function getReverseRangeByScore(float $min, float $max): Promise
+    public function getReverseRangeByScore(float $min, float $max, RangeByScoreOptions $options = null): Promise
```
Comment (kelunik) [inline src/RedisSortedSet.php:93]:
```suggestion
    public function getReverseRangeByScore(float $min, float $max, ?RangeByScoreOptions $options = null): Promise
```

**Comment by trowski** [comment]
Merged in https://github.com/amphp/redis/commit/b720d085e0ee765384e32cf7d0fc50ec0d75ba62.

**Pull request closed without merging**